### DISCUSSION TITLE
Feature Request: Tag Groups in Classic Layout

### DISCUSSION THREAD

**Discussion by Aderinom** [scalar/scalar#3765] [Ideas]
Hello Scalar Team! 

At the moment Scalar supports "x-tagGroups" by grouping operations in the Sidebar.
For the the classic layout as of now tag groups are not being rendered. 

It would be nice if tag groups would put a spoiler around the tags which are grouped inside of them in the classic view 

Similar to but better designed than: 
![image](https://github.com/user-attachments/assets/8d172208-3fde-46e1-9054-f73a9db1f2cc)

It would honestly be also nice if there was a way to do nested tags. E.g. making it possible to split tag names by some configureable delimiter e.g. "scope.my.tag.better" or using something more flexible than the tagGroups approach.

(edit: had a dangling sentence in the middle of the request)

**Comment by hanspagel** [comment]
You’re absolutely right! Issue here: https://github.com/scalar/scalar/issues/3766

**Reply by Aderinom** [reply]
That was fast haha

**Comment by nandgator** [comment]
Hi, I was looking for this!

How? I thought swagger UI didn't support multi-level or nested tags. Any reference?

How can I do this with Scalar and FastAPI?

**Reply by Aderinom** [reply]
The picture was taken out of a similar feature request from Swagger-UI.

There are one/some plugins for Swagger which would allow you to do this - don't know how well they work however. 
e.g. https://github.com/kael-shipman/swagger-ui-plugins/tree/hierarchical-tags-self-hosted/packages/hierarchical-tags

I don't think? that Scalar currently supports plugins in such a way.
